driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -339,30 +339,100 @@ static __always_inline uint16_t auxmap__store_bytebuf_param(struct auxiliary_map
 }
 
 /**
- * @brief Use `auxmap__store_execve_exe` when you have to store the
- * `exe` name from an execve-family syscall.
- * By convention, `exe` is `argv[0]`, this is the reason why here we pass the `argv` array.
+ * @brief This helper stores a char buffer array as a byte buffer into the auxmap.
  *
  * @param auxmap pointer to the auxmap in which we are storing the param.
- * @param array charbuf pointer array, obtained directly from the syscall (`argv`).
+ * @param start_pointer pointer where we start to read.
+ * @param len_to_read len that we can ideally read.
+ * @param max_len max len that we can read.
  */
-static __always_inline void auxmap__store_execve_exe(struct auxiliary_map *auxmap, char **array)
+static __always_inline void auxmap__store_charbufarray_as_bytebuf(struct auxiliary_map *auxmap, unsigned long start_pointer, uint16_t len_to_read, uint16_t max_len)
+{
+	/* Here we read an array of charbufs starting from a pointer. 
+	 * We could also read the array element per element but
+	 * since we know the total len we read it as a `bytebuf`.
+	 * Since this is an array of charbufs the `\0` after every argument are preserved.
+	 * We just need to add a final `\0` in case we args are too long and we have a partial
+	 * read. 
+	 */
+	if(len_to_read >= max_len)
+	{
+		len_to_read = max_len;
+	}
+
+	/* if `auxmap__store_bytebuf_param` returns 0 we will send an empty param.
+	 * we don't need the final `\0`.
+	 */
+	if(auxmap__store_bytebuf_param(auxmap, start_pointer, len_to_read, USER) > 0)
+	{
+		// maybe we read only part of the last argument so we need to put a `\0` at the end.
```

Comment:
if `auxmap__store_bytebuf_param` returned `0` we don't need to push a null terminator